### PR TITLE
feat: add static code diagnostic `prefer-using-list-view`

### PR DESCRIPTION
**Pull request by lsaudon**
### What is the purpose of this pull request? (put an "X" next to an item)

- [x] Documentation update
- [ ] Bug fix
- [x] New rule
- [ ] Changes an existing rule
- [ ] Add autofixing to a rule
- [ ] Add a CLI option
- [ ] Add something to the core
- [ ] Other, please explain:

- #400

### What changes did you make? (Give an overview)

Add static code diagnostic `prefer-using-list-view`
I check `SingleChildScrollView` with `child` `Column` with only `children` parameter.

### Is there anything you'd like reviewers to focus on?

Not to my knowledge.

### CODE REVIEW

File: test/src/analyzers/lint_analyzer/rules/rules_list/prefer_using_list_view/prefer_using_list_view_rule_test.dart [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+          ruleId: 'prefer-using-list-view',
+          severity: Severity.performance,
+        );
+      });
+
+      test('reports about found issues', () async {
+        final unit = await RuleTestHelper.resolveFromFile(_examplePath);
+        final issues = PreferUsingListViewRule().check(unit);
+
+        RuleTestHelper.verifyIssues(
```
Comment (incendial) [inline test/src/analyzers/lint_analyzer/rules/rules_list/prefer_using_list_view/prefer_using_list_view_rule_test.dart:28]: In the example file you have two places with the `LINT` comment, but there is only one issue. Which side is wrong? 🙂

File: test/src/analyzers/lint_analyzer/rules/rules_list/prefer_using_list_view/examples/example.dart [hunk 2]
```diff
@@ -0,0 +1,68 @@
+import 'package:flutter/material.dart';
+
+void main() {
+  runApp(MyApp());
+}
+
+class MyApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    // Lint
```
Comment (incendial) [inline test/src/analyzers/lint_analyzer/rules/rules_list/prefer_using_list_view/examples/example.dart:10]: Please change to uppercase (`LINT`)

File: website/docs/rules/flutter/prefer-using-list-view.mdx [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import RuleDetails from '@site/src/components/RuleDetails';
+
+<RuleDetails version="X.X.X" severity="performance" />
```
Comment (incendial) [inline website/docs/rules/flutter/prefer-using-list-view.mdx:3]: Please use 5.2.0 version here

File: website/docs/rules/index.mdx [hunk 4]
```diff
@@ -648,6 +648,15 @@ Rules are grouped by category to help you understand their purpose. Each rule ha
   Warns when a file contains more than a single widget.
 </RuleEntry>
 
+<RuleEntry
+  name="prefer-using-list-view"
+  type="flutter"
+  severity="performance"
+  version="X.X.X"
```
Comment (incendial) [inline website/docs/rules/index.mdx:655]: Same about the version

**Review by incendial — COMMENTED** [review]
@lsaudon thank you, great result! Left several small comments

File: test/src/analyzers/lint_analyzer/rules/rules_list/prefer_using_list_view/examples/example.dart (hunk 2 above)
Comment (lsaudon) [reply test/src/analyzers/lint_analyzer/rules/rules_list/prefer_using_list_view/examples/example.dart]: Copy & Paste is bad 😄

**Comment by incendial** [comment]
@lsaudon thank you!

**Pull request merged**